packages/stylex/src/StyleXSheet.js

```diff
@@ -91,13 +91,22 @@ export class StyleXSheet {
     this.injected = false;
     this.ruleForPriority = new Map();
     this.rules = [];
+    this.constantsCache = new Map();
 
     this.rootTheme = opts.rootTheme;
     this.rootDarkTheme = opts.rootDarkTheme;
     this.supportsVariables =
       opts.supportsVariables ?? doesSupportCSSVariables();
   }
 
+  constantsCache: Map<
+    string,
+    {|
+      value: string | number,
+      rules: Array<number | string>,
+    |},
```

Comment:
Object types are exact by default.

```suggestion
    {
      value: string | number,
      rules: Array<number | string>,
    },
```

Reply:
Also, the `rules` should either be an array of `number` (indexes) or `string`, but not both.

`string` *might* make a bit more sense as the indices are not stable.